### ISSUE TITLE
[bug] UndefinedConstant error

### ISSUE THREAD

**Issue by andrew-demb** [vimeo/psalm#1165]
If my class implements interface with constants, where exists constant assignment from another interface, psalm throws errors, but it's valid code.

https://getpsalm.org/r/4e822537cd

```php
<?php

interface Types {
    public const TWO = 'two';
}
  
interface A {
   public const TYPE_ONE = 'one';
   public const TYPE_TWO = Types::TWO;
}

class B implements A {
    public function __construct()
    {
      echo self::TYPE_ONE;
      echo self::TYPE_TWO; // this raises UndefinedConstant
    }
}
```

**Comment by muglug** [comment]
Thanks for logging!
